lib/data/repositories/year_repository.dart

```diff
@@ -54,7 +54,27 @@ class YearRepository extends CrudRepository<Year> {
         return Year.fromJson(data);
       }).toList();
       if (yearList.length > 1) {
-        throw MultipleElementError("Multiple Year found with id: $id");
+        throw MultipleElementException("Multiple Year found with id: $id");
+      }
+      return yearList.first;
+    }
+    return null;
+  }
+
+  Future<Year?> getByTenant(String tenantId) async {
+    QuerySnapshot<Map<String, dynamic>> yearsSnapshot = await firestore
+        .collection('years')
+        .where('tenantId', isEqualTo: tenantId)
+        .get();
+    if (yearsSnapshot.size > 0) {
+      List<Year> yearList = yearsSnapshot.docs.map((yearDoc) {
+        Map<String, dynamic> data = yearDoc.data();
+        data.putIfAbsent('id', () => yearDoc.id);
```

Comment:
There is no "id" field in "years", so this would always put documentID

Reply:
exactly, but the Map type doesn't have a simple `put` method the append an extra key-value pair into it, thus this has to happen to be able to create a proper Year object

Reply:
Ty, I misunderstood the aim of this.